lib/pbench/server/indexing_tarballs.py

```diff
@@ -267,6 +272,32 @@ def process_tb(self, tarballs):
         else:
             idxctx.set_tracking_id(tracking_id)
 
+        return res.value
+
+    def process_tb(self, tarballs: List[TarballData]) -> int:
+        """Process Tarballs For Indexing and creates report
+
+        Args:
+            tarballs:   List of tarball information tuples
+
+        Returns:
+            status code
+        """
+        res = 0
+        idxctx = self.idxctx
+        error_code = self.error_code
+
+        # Store the list of tarballs as a deque
+        tb_deque = deque(tarballs)
+
+        res = self.load_templates()
+        if res:
```

Comment:
Shouldn't this be `if res.success`?

Reply:
Well...later on `res` receives an `ErrorCode`, so that's probably what confused me.  (Yes, I should have paid better attention to the function signature return type hint.)

So, it would be good if `res` weren't reused like that (unless you're going to make `load_templates()` return an `ErrorCode`...).

What would also be good would be to viciously refactor `process_tb()`:  it is currently over **_400_** lines long (it occupies more than half of the modue!).  I just can't wait to see what the unit tests for it look like....  🤢

Reply:
I see that you've fixed this.  It is good, now...but `process_tb()` still need refactoring, and I'm still in trepidation over how we're gonna test it.